src/amr/data/particles/refine/split.h

```diff
@@ -12,6 +12,60 @@
 #include "core/data/particles/particle.h"
 #include "core/utilities/box/box.h"
 
+#include "kul/log.hpp"
+
+namespace PHARE::amr
+{
+template<size_t _dimension, size_t _refinedParticlesNbr>
+class ASplitter
+{
+public:
+    static constexpr size_t dimension           = _dimension;
+    static constexpr size_t refinedParticlesNbr = _refinedParticlesNbr;
+    static constexpr size_t refinementFactor    = 2;
+
+    inline void operator()(core::Particle<dimension> const& coarsePartOnRefinedGrid,
+                           std::vector<core::Particle<dimension>>& refinedParticles) const
+    {
+        auto get = [&](size_t rpIndex, size_t index, auto& icell, auto& delta) {
+            delta[index] += deltas_[rpIndex][index] * refinementFactor;
```

Comment:
I do not see where `deltas_` values are coming from... but from the approximate splitting (as well as the exact one), these `deltas_` values are already on the refine grid, so no need to multiply them by the `refinementFactor`

Reply:
this is an attempted generalisation of the existing operator()() @
https://github.com/Dekken/PHARE/blob/2d/src/amr/data/particles/refine/split.h#L257